### PR TITLE
Add case:repeatedly hotplug-unplg disks wildly without delay

### PR DESCRIPTION
**Pull request by qingwangrh**
ID:2185168
Signed-off-by: qingwangrh <qinwang@redhat.com>

### CODE REVIEW

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=multi_disk_wild_hotplug --guestname=RHEL.9.2.0 --clone=no --firmware=default_bios

(1/2) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.multi_disk_wild_hotplug.few_disks_without_delay.q35: STARTED
 (1/2) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.multi_disk_wild_hotplug.few_disks_without_delay.q35: ERROR: Login timeout expired    (output: 'exceeded 60 s timeout, last failure: Client said \'connection closed\'    (output: "Warning: Permanently added \'10.73.114.242\' (ED25519) to the list of known hosts.\\nConnection closed by 10.73.114.242 port 22\\n")') (325.39 s)
 (2/2) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.multi_disk_wild_hotplug.multi_disks_with_delay.q35: STARTED
 (2/2) Host_RHEL.m9.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.multi_disk_wild_hotplug.multi_disks_with_delay.q35: PASS (5152.84 s)
RESULTS    : PASS 1 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

Failed case is expected due to product bug not be fixed. it may pass RHEL8

**Comment by qingwangrh** [comment]
@YongxueHong @XueqiangWei @zhenyzha Please help to review, thanks.

File: qemu/tests/multi_disk_wild_hotplug.py [hunk 1]
```diff
@@ -34,15 +34,27 @@ def _run_sg_luns():
...
             params['image_name_%s' % name] = "images/%s" % name
             params['image_size_%s' % name] = stg_image_size
             params["images"] = params["images"] + " " + name
             if params["drive_format"] == "scsi-hd":
-                params["drive_bus_%s" % name] = 1
-                params["blk_extra_params_%s" % name] = "lun=%d" % i
+                if share_bus == "yes":
+                    params["drive_bus_%s" % name] = 0
+                else:
+                    params["drive_bus_%s" % name] = 1
```
[the remark this answers is not in the document]
Comment (qingwangrh) [reply qemu/tests/multi_disk_wild_hotplug.py:56]: update to params["drive_bus_%s" % name] = 0 if share_bus == "yes" else 1

File: qemu/tests/multi_disk_wild_hotplug.py [hunk 2]
```diff
@@ -101,9 +112,11 @@ def _hotunplug_images():
         _hotplug_images()
         time.sleep(plug_time)
 
+    error_context.context("Wait some time %d" % wait_time, test.log.info)
```
Comment (XueqiangWei) [inline qemu/tests/multi_disk_wild_hotplug.py:115]: Suggest adding time units, i.e. second here. 
e.g. Waiting for some time: 30 seconds.

Comment (qingwangrh) [reply qemu/tests/multi_disk_wild_hotplug.py]: updated, thanks

**Comment by XueqiangWei** [comment]
Tested with the default value of ulimit, hit an existing issue: qemu-kvm: virtio-blk failed to set guest notifier (-24), ensure -accel kvm is set. So  I think we should set a large value by parameter "vt_ulimit_nofile" in the test case. e.g. vt_ulimit_nofile = 65536

And then tested the cases, the results were as expected. Failed case is expected due to product bug not be fixed.

(1/4) Host_RHEL.m9.u3.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotes
t-qemu.multi_disk_wild_hotplug.without_delay.q35: PASS (308.72 s)
(2/4) Host_RHEL.m9.u3.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.multi_disk_wild_hotplug.with_delay.q35: PASS (1477.30 s)
(3/4) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.multi_disk_wild_hotplug.without_delay.q35: ERROR: Login timeout expired    (output: 'exceeded 60 s timeout, last failure: Client said \'connection closed\'    (output: "Warning: Permanently added \'10.73.211.169\' (ED25519) to the list of known hosts.\\nConnection closed by 10.73.211.169 port 22\\n")') (566.69 s)
(4/4) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.multi_disk_wild_hotplug.with_delay.q35: PASS (5142.06 s)

**Comment by qingwangrh** [comment]
> Tested with the default value of ulimit, hit an existing issue: qemu-kvm: virtio-blk failed to set guest notifier (-24), ensure -accel kvm is set. So I think we should set a large value by parameter "vt_ulimit_nofile" in the test case. e.g. vt_ulimit_nofile = 65536
> 
> And then tested the cases, the results were as expected. Failed case is expected due to product bug not be fixed.
> 
> (1/4) Host_RHEL.m9.u3.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotes t-qemu.multi_disk_wild_hotplug.without_delay.q35: PASS (308.72 s) (2/4) Host_RHEL.m9.u3.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.multi_disk_wild_hotplug.with_delay.q35: PASS (1477.30 s) (3/4) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.multi_disk_wild_hotplug.without_delay.q35: ERROR: Login timeout expired (output: 'exceeded 60 s timeout, last failure: Client said 'connection closed' (output: "Warning: Permanently added '10.73.211.169' (ED25519) to the list of known hosts.\nConnection closed by 10.73.211.169 port 22\n")') (566.69 s) (4/4) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.multi_disk_wild_hotplug.with_delay.q35: PASS (5142.06 s)

updated

**Comment by zhenyzha** [comment]
Thanks for the effort, this case can be a good way to check product issues.

LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

```
 (1/2) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.io-github-autotest-qemu.multi_disk_wild_hotplug.without_delay.arm64-pci: STARTED
 (1/2) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.io-github-autotest-qemu.multi_disk_wild_hotplug.without_delay.arm64-pci: ERROR: Login timeout expired    (output: 'exceeded 60 s timeout, last failure: Client said \'connection closed\'    (output: "Warning: Permanently added \'192.168.122.162\' (ED25519) to the list of known hosts.\\nConnection closed by 192.168.122.162 port 22\\n")... (894.33 s)
 (2/2) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.io-github-autotest-qemu.multi_disk_wild_hotplug.with_delay.arm64-pci: STARTED
 (2/2) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.io-github-autotest-qemu.multi_disk_wild_hotplug.with_delay.arm64-pci: PASS (5368.05 s)
```

**Review by zhenyzha — APPROVED** [review]

**Review by XueqiangWei — APPROVED** [review]

**Review by YongxueHong — APPROVED** [review]
LGTM.

**Pull request merged**